arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/TxSkipSection.java

```diff
@@ -84,80 +87,119 @@ public TxSkipSection(
-    final Address recipientAddress = txMetadata.getEffectiveRecipient();
-    final Address coinbaseAddress = txMetadata.getCoinbase();
-
+  public void resolvePostTransaction(Hub hub, WorldView world, Transaction tx, boolean statusCode) {
     checkArgument(statusCode, "TX_SKIP transactions should be successful");
-    checkArgument(
-        statusCode == txMetadata.statusCode(), "meta data suggests an unsuccessful TX_SKIP");
-
-    final AccountSnapshot senderAccountSnapshotAfter =
-        AccountSnapshot.fromAccount(
-            state.get(senderAddress),
-            this.senderAccountSnapshotBefore.isWarm(),
-            hub.deploymentNumberOf(senderAddress),
-            false);
-
-    final AccountSnapshot recipientAccountSnapshotAfter =
-        AccountSnapshot.fromAccount(
-            state.get(recipientAddress),
-            this.recipientAccountSnapshotBefore.isWarm(),
-            hub.deploymentNumberOf(recipientAddress),
-            false);
-
-    final AccountSnapshot coinbaseAccountSnapshotAfter =
-        AccountSnapshot.fromAccount(
-            state.get(coinbaseAddress),
-            this.coinbaseAccountSnapshotBefore.isWarm(),
-            hub.deploymentNumberOf(coinbaseAddress),
-            false);
-
-    // sender account fragment
-    final AccountFragment senderAccountFragment =
-        hub.factories()
-            .accountFragment()
-            .makeWithTrm(
-                senderAccountSnapshotBefore,
-                senderAccountSnapshotAfter,
-                senderAddress,
-                DomSubStampsSubFragment.standardDomSubStamps(hub.stamp(), 0));
-
-    // recipient account fragment
-    final AccountFragment recipientAccountFragment =
-        hub.factories()
-            .accountFragment()
-            .makeWithTrm(
-                recipientAccountSnapshotBefore,
-                recipientAccountSnapshotAfter,
-                recipientAddress,
-                DomSubStampsSubFragment.standardDomSubStamps(hub.stamp(), 1));
-
-    // coinbase account fragment
-    final AccountFragment coinbaseAccountFragment =
-        hub.factories()
-            .accountFragment()
-            .makeWithTrm(
-                coinbaseAccountSnapshotBefore,
-                coinbaseAccountSnapshotAfter,
-                coinbaseAddress,
-                DomSubStampsSubFragment.standardDomSubStamps(hub.stamp(), 2));
+    checkArgument(txMetadata.statusCode(), "meta data suggests an unsuccessful TX_SKIP");
+
+    if (txMetadata.noAddressCollisions()) {
+
+      final AccountSnapshot senderAccountSnapshotAfter =
+          AccountSnapshot.canonical(hub, world, senderAccountSnapshotBefore.address());
+
+      final AccountSnapshot recipientAccountSnapshotAfter =
+          AccountSnapshot.canonical(hub, world, recipientAccountSnapshotBefore.address());
+
+      final AccountSnapshot coinbaseAccountSnapshotAfter =
+          AccountSnapshot.canonical(hub, world, coinbaseAccountSnapshotBefore.address());
```

Comment:
I don't think IRL it's possible to have sender / coinbase that is a precompile ...

In the spec, we only constrain the warmth to remains constant, thus it's better to be hot -> hot as it could be warmed (being on the accessList , or so). So when taking teh snapshot, better to turn on the warmth all the time.